packages/lib-classifier/src/store/UserProjectPreferencesStore.js

```diff
@@ -89,8 +92,30 @@ const UserProjectPreferencesStore = types
       }
     }
 
-    function updateUPP (changes) {
-      console.log('TODO: UPP PUT request', changes)
+    function * updateUPP (changes) {
+      const { authClient } = getRoot(self)
+      const upp = self.resources.get(self.active.id)
+      if (upp) {
+        const authorization = yield getBearerToken(authClient)
+        const { type } = self
+        const client = getRoot(self).client.panoptes
+        self.loadingState = asyncStates.putting
+        try {
+          const newUPP = merge({}, upp, changes)
```

Comment:
I _think_ you can PUT the changeset, and get back the updated resource from the latest version of Panoptes.

Reply:
paging @camallen for confirmation  ^

Reply:
I think Pandora puts partial changes on save, which is where I first came across this.

Reply:
You can partially update the non JSON fields. our version of postgres doesn't support partial JSON updates (newer versions do but the API doesn't).

If you are updating the JSON `preferences` field then send the whole object with new changes in the PUT.

> Editing the preferences field requires a full representation of the preferences hash to be sent.

https://panoptes.docs.apiary.io/#reference/projectpreference/projectpreference/edit-a-projectpreference

Reply:
That makes sense, and is also the reason that tutorials and field guides are hard to update with multiple users in the project builder at the same time. IIRC, field guide content is just one big JSON blob?

Reply:
Yeah - items in the field guide is json attribute. For ref, we're on 9.4, this function is 9.5 https://wiki.postgresql.org/wiki/What's_new_in_PostgreSQL_9.5#jsonb_set_function

> The above lets us delete values in a path, but not update them, so we have the jsonb_set function for that. We'll update the phone value within the contact value: